lib/httpserver/httpserver.go

```diff
@@ -27,6 +27,7 @@ var (
 	tlsCertFile = flag.String("tlsCertFile", "", "Path to file with TLS certificate. Used only if -tls is set. Prefer ECDSA certs instead of RSA certs, since RSA certs are slow")
 	tlsKeyFile  = flag.String("tlsKeyFile", "", "Path to file with TLS key. Used only if -tls is set")
 
+	httpExternalURL  = flag.String("http.externalURL", "", "The URL under which the http service is externally reachable")
```

Comment:
Could you provide use case for this feature? Usually url rewriting should be performed by a proxy sitting in front of VictoriaMetrics, so the resulting url that is sent to VictoriaMetrics shouldn't contain any additional prefixes.

Reply:
just to make sure behavior same as Prometheus, because we are replacing Prometheus with vmagent, and here we have no proxy before Prometheus, so such a param is needed

Reply:
vm is a monorepo and httpserver packages is used in vmselect, vmalert, vmagent ... so this flag appears in all services, what we definitely try to avoid. Please add flag direct to vmagent
example you may find in vmalert package, it already supports the externalURL flag.

Reply:
The `external.url` in `vmagent` isn't related to the `http.externalURL` here. I think @xbsura pointed to valid reason to add `http.externalURL` to all the binaries that import `lib/httpserver`, since makes their management more uniform.

So I'm going to merge this PR. The only thing I don't like is `http.externalURL` flag name, which doesn't reflect the purpose of this flag. I'm going to rename it to `http.pathPrefix`

Reply:
thanks!